src/main/java/com/hierynomus/smbj/share/DiskShare.java

```diff
@@ -434,17 +565,115 @@ public void setSecurityInfo(SMB2FileId fileId, Set<SecurityInformation> security
         );
     }
 
+    /***
+     * 3.2.5.19 Receiving an SMB2 OPLOCK_BREAK Notification, this handler is responsible to call acknowledgeOplockBreak if needed. Set the handler for Receiving an Oplock Break Notification.
+     * You MUST set this handler before create/open diskEntry with oplock.
+     *
+     * @param handler handler for Receiving an Oplock Break Notification and Async Create Request/Response.
+     */
+    public void setNotificationHandler(NotificationHandler handler) {
+        this.notificationHandler = handler;
+    }
+
+    /***
+     * Handler for handing the oplock break notification event from server. 3.2.5.19 Receiving an SMB2 OPLOCK_BREAK Notification.
+     *
+     * @param oplockBreakNotification received oplock break notification from server.
+     */
+    @Handler
+    @SuppressWarnings("unused")
+    private void oplockBreakNotification(final OplockBreakNotification oplockBreakNotification) {
+        try {
+
+            final SMB2FileId fileId = oplockBreakNotification.getFileId();
+            final SMB2OplockBreakLevel oplockLevel = oplockBreakNotification.getOplockLevel();
+            logger.debug("FileId {} received OplockBreakNotification, Oplock level {}", fileId, oplockLevel);
+
+            if(notificationHandler != null) {
+                // Preventing the improper use of handler (holding the thread). if holding thread, timeout exception will be throw.
+                taskQueue.execute(new Runnable() {
+                    @Override
+                    public void run() {
+                        notificationHandler.handleOplockBreakNotification(oplockBreakNotification);
+                    }
+                }, notifyExecutor);
+            }else {
+                logger.warn("FileId {}, NotificationHandler not exist to handle Oplock Break.", fileId);
+                throw new IllegalStateException("NotificationHandler not exist to handle Oplock Break.");
```

Comment:
Why is this an exception, and the `createRequestNotification` does not throw an exception?

Reply:
Because the `createRequestNotification` and `createResponseNotification` is always called even you are using synchronize `create` (the existing `create`). The is a valid use case for a user to using the synchronize `create` and not set the NotificationHandler (as this is intended to handle async event only).

However, it is not a valid use case for using any level of oplock higher than `None` on any kind of `create` command. Because the `OplockBreakNotification` is always "Async" (or I should say Server initiated?).